chrome/content/zotero/xpcom/openurl.js

```diff
@@ -203,25 +203,33 @@ Zotero.OpenURL = new function() {
 			}
 		}
 		
-		if(item.creators && item.creators.length) {
-			// encode first author as first and last
-			var firstCreator = item.creators[0];
-			if(item.itemType == "patent") {
-				_mapTag(firstCreator.firstName, "invfirst");
-				_mapTag(firstCreator.lastName, "invlast");
-			} else {
-				if(firstCreator.isInstitution) {
-					_mapTag(firstCreator.lastName, "aucorp");
-				} else {
-					_mapTag(firstCreator.firstName, "aufirst");
-					_mapTag(firstCreator.lastName, "aulast");
+		var first = true;
+		for(var i=0; i<item.creators.length; i++) {
+			var creator = item.creators[i];
+			// only export authors
+			// other creator types cannot correctly export to COinS
+			if(creator.creatorType == "author") {
+				// encode first author as aufirst and aulast
+				if (first) {
+					if(item.itemType == "patent") {
+						_mapTag(creator.firstName, "invfirst");
+						_mapTag(creator.lastName, "invlast");
+					} else {
+						_mapTag(creator.firstName, "aufirst");
+						_mapTag(creator.lastName, "aulast");
+					}
```

Comment:
Yeah, that's what the spec would want you to believe, but I don't think the
alternatives make sense. Let's see what @adam3smith thinks
On Aug 3, 2015 12:48 AM, "Philipp Zumstein" <notifications@github.com>
wrote:

> In chrome/content/zotero/xpcom/openurl.js
> <https://github.com/aurimasv/zotero/pull/27#discussion_r36057567>:
>
> > -					_mapTag(firstCreator.lastName, "aulast");
> > +		var first = true;
> > +		for(var i=0; i<item.creators.length; i++) {
> > +			var creator = item.creators[i];
> > +			// only export authors
> > +			// other creator types cannot correctly export to COinS
> > +			if(creator.creatorType == "author") {
> > +				// encode first author as aufirst and aulast
> > +				if (first) {
> > +					if(item.itemType == "patent") {
> > +						_mapTag(creator.firstName, "invfirst");
> > +						_mapTag(creator.lastName, "invlast");
> > +					} else {
> > +						_mapTag(creator.firstName, "aufirst");
> > +						_mapTag(creator.lastName, "aulast");
> > +					}
>
> I read in the OpenURL documentation
> <http://www.niso.org/kst/reports/standards/kfile_download?id%3Austring%3Aiso-8859-1=Z39-88-2004.pdf&pt=RkGKiXzW643YeUaYUqZ1BFwDhIG4-24RJbcZBWg8uE4vWdpZsJDs4RjLz0t90_d5_ymGsj_IKVaGZww13HuDlZQ8NBt1sTxP_v4iiGqH7rSaAeVDnMfeKJrrJ_JSEGPB>
> (p. 37) that aucorp can occur at most once:
>
> [image: openurl-aucorp]
> <https://cloud.githubusercontent.com/assets/5199995/9031440/50edad68-39b3-11e5-9b2e-a3bebe753049.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aurimasv/zotero/pull/27/files#r36057567>.
>